### PR TITLE
lint: add stylelint support for emotion components

### PR DESCRIPTION
**Pull request by uladzimirdev**
Successor of https://github.com/metabase/metabase/pull/33043

### Description

Add stylelint support for emotion/styled components

### How to verify

CI is green

### CODE REVIEW

File: frontend/src/metabase/core/components/AccordionList/AccordionListCell.styled.tsx [hunk 1]
```diff
@@ -21,11 +21,6 @@ export const Content = styled.div<{ isClickable: boolean }>`
   padding: 0.5rem;
   cursor: ${props => (props.isClickable ? "pointer" : "default")};
   min-width: 0;
-
-  [data-element-id="list-item"]-content {
```
Comment (uladzimirdev) [inline frontend/src/metabase/core/components/AccordionList/AccordionListCell.styled.tsx:25]: it's not used anymore, [confirmed](https://metaboat.slack.com/archives/C0645JP1W81/p1720097398102519)

File: frontend/src/metabase/visualizations/visualizations/Text/Text.styled.tsx [hunk 2]
```diff
@@ -229,9 +236,6 @@ export const ReactMarkdownStyleWrapper = styled.div`
     text-decoration: none;
     color: var(--mb-color-brand);
   }
-  .text-card-markdown a:hover {
```
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/visualizations/Text/Text.styled.tsx:232]: duplicated below

File: frontend/src/metabase/components/EntityMenuItem/EntityMenuItem.styled.tsx [hunk 3]
```diff
@@ -22,39 +22,46 @@ export const MenuItemContent = styled.div<MenuItemProps>`
...
+
   :hover > .Icon {
     color: ${props => color((!props.disabled && props.hoverColor) || "brand")};
-  },
+  }
+
   /* icon specific tweaks
      the alert icon should be optically aligned  with the x-height of the text */
   > .Icon.Icon-alert {
-    transform: translate-y(1px),
```
Comment (alxnddr) [inline frontend/src/metabase/components/EntityMenuItem/EntityMenuItem.styled.tsx:43]: Did this work?

**Review by alxnddr — APPROVED** [review]
I left one comment, please have a look before merging

File: frontend/src/metabase/components/EntityMenuItem/EntityMenuItem.styled.tsx (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/components/EntityMenuItem/EntityMenuItem.styled.tsx:43]: It didn't, it's not a valid syntax. Probably safer to drop it

File: frontend/src/metabase/browse/components/ModelsTable.styled.tsx [hunk 4]
```diff
@@ -9,8 +9,8 @@ import type { ResponsiveProps } from "metabase/components/ItemsTable/utils";
 import { breakpoints } from "metabase/ui/theme";
 
 export const ModelTableRow = styled.tr<{ skeleton?: boolean }>`
-  :outline {
-    outline: 2px solid var(--mb-color-brand);
+  :focus {
+    outline: 2px solid var(--mb-color-focus);
   }
```
Comment (heypoom) [inline frontend/src/metabase/browse/components/ModelsTable.styled.tsx:14]: Is the `--mb-color-focus` change expected?

**Review by heypoom — APPROVED** [review]
Diffs should be safe to merge. Nice!

File: frontend/src/metabase/browse/components/ModelsTable.styled.tsx (hunk 4 above)
Comment (uladzimirdev) [reply frontend/src/metabase/browse/components/ModelsTable.styled.tsx:14]: yes, it's made by the code author

**Review by oisincoveney — APPROVED** [review]

File: frontend/src/metabase/components/EntityMenuItem/EntityMenuItem.styled.tsx (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/components/EntityMenuItem/EntityMenuItem.styled.tsx:43]: [`b2a9499` (#45132)](https://github.com/metabase/metabase/pull/45132/commits/b2a9499bbb060a57978bae05cf0e91bcbf2caf90)

**Pull request merged**